soc/arm/nxp_imx/rt/soc_rt10xx.c

```diff
@@ -135,11 +135,14 @@ static ALWAYS_INLINE void clock_init(void)
 	CLOCK_InitVideoPll(&videoPllConfig);
 #endif
 
-#ifdef CONFIG_HAS_ARM_DIV
-	CLOCK_SetDiv(kCLOCK_ArmDiv, CONFIG_ARM_DIV); /* Set ARM PODF */
+#if DT_NODE_EXISTS(DT_CHILD(CCM_NODE, arm_podf))
```

Comment:
Could you use BUILD_ASSERT here for each clock to check the range information that was previously encoded via KConfig? I'd like to preserve this as it makes it easier for the user to avoid an invalid clock configuration, IMO.

Reply:
To add to this, I am also OK with use comments in the DTS to document this. Just want to make it easier for developers to find this information. Clock changes themselves LGTM, I verified this PR on all platforms except the RT1064 and RT1020

Reply:
@danieldegrasse I've followed your first suggestion: https://github.com/zephyrproject-rtos/zephyr/pull/49782#issuecomment-1290269083